### PR TITLE
Add 2.4 api

### PR DESCRIPTION
**Pull request by yuanzhel**
implement vsync2.4 callback, setactiveconfigwithconstraints
and getdisplayvsyncperiod.

Add an empty SetDisplayBrightness to resolove boot problem

Tracked-On: OAM-92711
Signed-Off-By: Liu, Yuanzhe <yuanzhe.liu@intel.com>

### CODE REVIEW

File: common/display/vblankeventhandler.cpp [hunk 1]
```diff
@@ -87,12 +97,19 @@ void VblankEventHandler::HandlePageFlipEvent(unsigned int sec,
   int64_t timestamp = ((int64_t)sec * kOneSecondNs) + ((int64_t)usec * 1000);
   IPAGEFLIPEVENTTRACE("HandleVblankCallBack Frame Time %f",
                       static_cast<float>(timestamp - last_timestamp_) / (1000));
+  double vperiod = static_cast<double>(timestamp - last_timestamp_) / (1000));
```
Comment (Shao-Feng) [inline common/display/vblankeventhandler.cpp:100]: Should not use double float in comparing.

File: common/display/vblankeventhandler.cpp [hunk 2]
```diff
@@ -87,12 +97,19 @@ void VblankEventHandler::HandlePageFlipEvent(unsigned int sec,
...
   last_timestamp_ = timestamp;
 
   IPAGEFLIPEVENTTRACE("Callback called from HandlePageFlipEvent. %lu",
                       timestamp);
   spin_lock_.lock();
   if (enabled_ && callback_) {
+     if(abs(vperiod - vperiod_) > (1/90 - 1/75))
```
Comment (Shao-Feng) [inline common/display/vblankeventhandler.cpp:107]: Should not use double float in comparing.

Comment (Shao-Feng) [reply common/display/vblankeventhandler.cpp:107]: 1/90 is 0 and 1/75 is also 0

Comment (Shao-Feng) [reply common/display/vblankeventhandler.cpp:107]: should use long int for comparing and calculating

File: common/display/vblankeventhandler.cpp [hunk 3]
```diff
@@ -87,12 +97,19 @@ void VblankEventHandler::HandlePageFlipEvent(unsigned int sec,
...
+      callback_2_4_->Callback(display_, timestamp, static_cast<uint32_t>(vperiod * 1000 * 1000));
+     else
+      callback_->Callback(display_, timestamp);
```
Comment (Shao-Feng) [inline common/display/vblankeventhandler.cpp:110]: need to check if the original Callback still need to be invoked, or it is replaced by the new one.

**Review by Shao-Feng — CHANGES_REQUESTED** [review]

File: os/android/iahwc2.cpp [hunk 4]
```diff
@@ -815,6 +855,25 @@ HWC2::Error IAHWC2::HwcDisplay::SetActiveConfig(hwc2_config_t config) {
...
+  hwc_vsync_period_change_constraints_t* vsyncPeriodChangeConstraints,
+  hwc_vsync_period_change_timeline_t* outTimeline
+  ) {
+  supported(__func__);
+  return SetActiveConfig(config);
+}
+
+HWC2::Error IAHWC2::HwcDisplay::GetDisplayVsyncPeriod(hwc2_vsync_period_t* outVsyncPeriod) {
+  supported(__func__);
+  *outVsyncPeriod = 16666667;
```
Comment (YuanjunHuang) [inline os/android/iahwc2.cpp:869]: Use a Macro like DEFAULT_VSYNC_PERIOD defined in head file

File: common/display/vblankeventhandler.cpp [hunk 5]
```diff
@@ -87,14 +97,19 @@ void VblankEventHandler::HandlePageFlipEvent(unsigned int sec,
...
                       static_cast<float>(timestamp - last_timestamp_) / (1000));
+  int64_t vperiod = timestamp - last_timestamp_;
   last_timestamp_ = timestamp;
 
   IPAGEFLIPEVENTTRACE("Callback called from HandlePageFlipEvent. %lu",
                       timestamp);
   spin_lock_.lock();
   if (enabled_ && callback_) {
-    callback_->Callback(display_, timestamp);
+     if(abs(vperiod - vperiod_) > (13333333 - 11111111))
```
Comment (YuanjunHuang) [inline common/display/vblankeventhandler.cpp:107]: replace with Macro like 90Hz_PERIOD_NANOS and 75Hz_PERIOD_NANOS, the number is hard to understand

File: os/android/iahwc2.cpp (hunk 4 above)
Comment (Shao-Feng) [reply os/android/iahwc2.cpp:869]: why the vsync period need to be initialized with default here?
no need to do that, refer to https://github.com/intel/IA-Hardware-Composer/blob/master/wsi/drm/drmdisplay.cpp#L273

File: wsi/drm/drmdisplay.cpp [hunk 6]
```diff
@@ -575,9 +575,20 @@ void DrmDisplay::UpdateDisplayConfig() {
...
   SetDisplayAttribute(modes_[config_]);
+  vsync_period_ = modes_[config_].vrefresh;
   SPIN_UNLOCK(display_lock_);
 }
 
+bool DrmDisplay::GetDisplayVsyncPeriod(uint32_t* outVsyncPeriod) {
+  SPIN_LOCK(display_lock_);
+  if (modes_.empty()) {
+    SPIN_UNLOCK(display_lock_);
+    return false;
```
Comment (Shao-Feng) [inline wsi/drm/drmdisplay.cpp:586]: if modes is empty, you can return the default one.
it is how we handle headless case https://github.com/intel/IA-Hardware-Composer/blob/master/wsi/drm/drmdisplay.cpp#L273

File: public/hwcutils.h [hunk 7]
```diff
@@ -25,7 +25,7 @@
 #include <sstream>
 #include "overlaylayer.h"
 
-#define ALL_EDID_FLAG_PROPERTY "vendor.hwcomposer.edid.all"
+#define ALL_EDID_FLAG_PROPERTY "1"
```
Comment (Shao-Feng) [inline public/hwcutils.h:28]: why do you change this one?, if you need a workaround, you can uncomment it in invoking

**Review by Shao-Feng — CHANGES_REQUESTED** [review]
need to follow hwc coding style for all parameter, variable and method names

**Pull request merged**